crates/astria-sequencer-relayer/tests/blackbox/helper.rs

```diff
@@ -305,6 +326,12 @@ pub struct TestSequencerRelayerConfig {
 
 impl TestSequencerRelayerConfig {
     pub async fn spawn_relayer(self) -> TestSequencerRelayer {
+        assert_ne!(
```

Comment:
We need the tokio runtime to not be of the "current thread" flavour so as to allow the `TestSequencerRelayer::drop` to not hang at the `futures::executor::block_on` call.

Reply:
Well, it's not a problem with code currently in the `main` branch.  It's the addition in this PR of the call to `futures::executor::block_on` in the `TestSequencerRelayer::drop`.

The async function passed to `block_on` can't be cancelled or paused, and I believe that's stopping the relayer making progress inside the main `select!` macro as the single tokio thread is blocked.